core/common/src/main/java/alluxio/util/JvmPauseMonitor.java

```diff
@@ -0,0 +1,213 @@
+        Map<String, GarbageCollectorMXBean> gcMXBeanMapAfterSleep) {
+    List<String> diffBean = Lists.newArrayList();
+    GarbageCollectorMXBean oldBean;
+    GarbageCollectorMXBean newBean;
+    Set<String> nameSet = Sets.intersection(gcMXBeanMapBeforeSleep.keySet(),
+        gcMXBeanMapAfterSleep.keySet());
+    for (String name : nameSet) {
+      oldBean = gcMXBeanMapBeforeSleep.get(name);
+      newBean = gcMXBeanMapAfterSleep.get(name);
+      if (oldBean == null) {
+        diffBean.add("new GCBean created name= '" + newBean.getName() + " count="
+            + newBean.getCollectionCount() + " time=" + newBean.getCollectionTime() + "ms");
+      } else if (newBean == null) {
+        diffBean.add("old GCBean canceled name= '" + oldBean.getName() + " count="
+            + oldBean.getCollectionCount() + " time=" + oldBean.getCollectionTime() + "ms");
+      } else {
+        if (oldBean.getCollectionTime() != newBean.getCollectionTime()
+            || oldBean.getCollectionCount() != newBean.getCollectionCount()) {
+          diffBean.add("GC name= '" + newBean.getName() + " count="
+              + newBean.getCollectionCount() + " time=" + newBean.getCollectionTime() + "ms");
+        }
+      }
+    }
+    String ret = "JVM paused " + extraSleepTime + "ms\n";
+    if (diffBean.isEmpty()) {
+      ret += "No GCs detected ";
+    } else {
+      ret += "GC list:\n" + Joiner.on("\n").join(diffBean);
+    }
+    ret += "\n" + getMemoryInfo();
+    return ret;
+  }
+
+  private Map<String, GarbageCollectorMXBean> getGarbageCollectorMXBeans() {
+    List<GarbageCollectorMXBean> gcBeanList = ManagementFactory.getGarbageCollectorMXBeans();
+    Map<String, GarbageCollectorMXBean> gcBeanMap = new HashMap();
+    for (GarbageCollectorMXBean gcBean : gcBeanList) {
+      gcBeanMap.put(gcBean.getName(), gcBean);
+    }
+    return gcBeanMap;
+  }
+
+  private class Monitor implements Runnable {
+    @Override
+    public void run() {
+      Stopwatch sw = new Stopwatch();
+      Map<String, GarbageCollectorMXBean> gcBeanMapBeforeSleep = getGarbageCollectorMXBeans();
+      while (true) {
+        sw.reset().start();
+        try {
+          Thread.sleep(mGcSleepIntervalMs);
+        } catch (InterruptedException ie) {
+          LOG.warn(ie.getStackTrace());
+          return;
+        }
+        long extraTime = sw.elapsed(TimeUnit.MILLISECONDS) - mGcSleepIntervalMs;
+        mTotalExtraTimeMs += extraTime;
+        Map<String, GarbageCollectorMXBean> gcBeanMapAfterSleep = getGarbageCollectorMXBeans();
+
+        if (extraTime > mWarnThresholdMs) {
+          ++mWarnTimeExceededMS;
+          LOG.warn(formatLogString(
+              extraTime, gcBeanMapBeforeSleep, gcBeanMapAfterSleep));
+        } else if (extraTime > mInfoThresholdMs) {
+          ++mInfoTimeExceededMS;
+          LOG.info(formatLogString(
+              extraTime, gcBeanMapBeforeSleep, gcBeanMapAfterSleep));
+        }
```

Comment:
Should we increment both info and warn if it exceeds warn?

Reply:
@calvinjia  I don not think the increment of both info and warn is necessary. Because the only differents between them is the log level.

Reply:
If I understand correctly, with the current implementation:
If info threshold is 10 seconds, and warn is 100 seconds, if a pause is 150 seconds, `mInfoTimeExceeded` will be `0` and `mWarnTimeExceeded` will be `1`.

Based on the definition, it seems `mInfoTimeExceeded` should always be greater than or equal to `mWarnTimeExceeded`.